src/v/storage/segment_utils.cc

```diff
@@ -491,4 +493,17 @@ std::filesystem::path compacted_index_path(std::filesystem::path segment_path) {
     return segment_path.replace_extension(".compaction_index");
 }
 
+size_t jitter_segment_size(size_t sz, jitter_percents jitter_percents) {
+    vassert(
+      jitter_percents >= 0 || jitter_percents <= 100,
+      "jitter percents should be in range [0,100]. Requested {}",
+      jitter_percents);
+    // multiply by 10 to increase resolution
+    auto p = jitter_percents() * 10;
+    auto jit = random_generators::get_int<int64_t>(-p, p);
+
+    int64_t jitter = jit * static_cast<int64_t>(sz) / 1000;
+    return jitter + sz;
```

Comment:
this should probably get clamped by a lower bound > 0 since as is this could go negative.

Reply:
good point.

Reply:
good point, sending fix for this

Reply:
I will add a `vassert` here as for this to happen segment would have to be millions of terabytes in size.

Reply:
wouldn't it only take a jitter pct of 100%?

Reply:
vassert(
      jitter_percents >= 0 || jitter_percents <= 100,
      "jitter percents should be in range [0,100]. Requested {}",
      jitter_percents);